pkg/skaffold/parser/config.go

```diff
@@ -164,7 +164,7 @@ func processEachConfig(config *latest_v1.SkaffoldConfig, cfgOpts configOpts, opt
 
 	var configs []*latest_v1.SkaffoldConfig
 	for _, d := range config.Dependencies {
-		newOpts := configOpts{file: cfgOpts.file, profiles: filterActiveProfiles(d, profiles), isRequired: required, isDependency: cfgOpts.isDependency}
+		newOpts := configOpts{file: cfgOpts.file, profiles: filterActiveProfiles(d, profiles, opts), isRequired: required, isDependency: cfgOpts.isDependency}
```

Comment:
`filterActiveProfiles()` should also consider the propagated profiles too, right?  So maybe it makes more sense to build up the list of active profiles here (`profiles` + `opts.Profiles` if propagating), and then add the results of `filterActiveProfiles()` which is really just looking for activated profiles?  

Something like
```suggestion
		active := profiles
		if opts.ApplyProfilesRecursively {
			for _, p := range opts.Profiles {
				if !util.StrSliceContains(active, p) {
					active = append(active, p)
				}
			}
		}
		for _, p := range filterActiveProfiles(d, active) {
			if !util.StrSliceContains(active, p) {
				active = append(active, p)
			}
		}
		newOpts := configOpts{file: cfgOpts.file, profiles: active, isRequired: required, isDependency: cfgOpts.isDependency}
```